### PR TITLE
 BUG  : Bird classification problem :replaced deprecated keras.preprocessing.image_data_generator with tf.…

### PR DESCRIPTION
**Pull request by yashashwanidixit**
The original code uses deprecated keras.preprocessing.ImageDatagenerator and flow_from_directory which is not supported on latest keras version.I have replaced keras.preprocessing.flow_from_directory with tensorflow.keras.utils.image_dataset_from_directory and used tf.keras.sequential for data augmentation.
This change is necessary as the original keras.preprocessing.Imagedatagenerator is not supported on the latest keras version.

### CODE REVIEW

**Comment by yashashwanidixit** [comment]
@Niketkumardheeryan  sir,
Could u please merge the changes ,the changes are minor but important .Also if you find anything that has to be improved or something that has to be changed,please inform and i will definitely make the required changes.
Thankyou very much

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
@yashashwanidixit can u run all the cells please

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
please run these rerun the cells every single one of them 

Image

Image

there cells haven't been run 
@yashashwanidixit @yashashwanidixit

**Comment by yashashwanidixit** [comment]
So sorry for the delay—I missed this notification. I'll run all the cells and update the notebook in the next few hours.

**Comment by yashashwanidixit** [comment]
Sorry for the late response. I tried running all the cells, but Google Colab keeps disconnecting repeatedly, so I haven't been able to complete the execution. I also don't have access to a GPU on my laptop, so running everything locally would take several hours(we have 2645 cycles per epoch and 10 epochs ). If it's okay, I can push or share  the latest code with all the changes, and you could run the cells on your end. Really sorry for the delay and the inconvenience.

**Review by Sibam-Paul — APPROVED** [review]
ok this will be accepted

**Comment by Sibam-Paul** [comment]
@yashashwanidixit u can use google cloud console

**Comment by yashashwanidixit** [comment]
Alright, i will try to run this on google cloud console and keep you updated on the progress.

**Comment by yashashwanidixit** [comment]
@Sibam-Paul I ran into a dependency issue while setting up the notebook on vertex ai workbench. the project requires tensorflow==2.8.0, but the default environment uses Python 3.12, which is not compatible with TensorFlow 2.8. i am  testing with updated dependency versions to see if the notebook runs successfully without breaking compatibility, and I'll share the results once I have them.But i will be better to change the requirements.txt since the tf version is not compatible with current python versions. Please allow for the same.

**Comment by Sibam-Paul** [comment]
Ok raise another one @yashashwanidixit

**Comment by yashashwanidixit** [comment]
i have raised please assign

**Comment by Sibam-Paul** [comment]
Self assign with /assign

**Comment by yashashwanidixit** [comment]
Screenshot 2026-07-06 173743

i am getting a timeout  error during checks for pull requests...ig there is something wrong with Copy_of_FineTuned_LLama_Medical_Terms.ipynb

**Comment by yashashwanidixit** [comment]
@Sibam-Paul RAN ALL THE CELLS final accuracy is 91 % . I tried to create a pull request but i am getting this error whilst -




image
Please guide .

**Comment by Sibam-Paul** [comment]
@yashashwanidixit there are notebook in the repo which is not in good condition I had made a issue about it u can see
